### DISCUSSION TITLE
Keycloak 19 as deployment or statefulset?

### DISCUSSION THREAD

**Discussion by vishrane** [keycloak/keycloak#16716] [Q&A]
Hi Team,
     We are creating keycloak image using base image quay.io/keycloak/keycloak:19.0.2. Then, we tried deploying keycloak using deployment and statefulset. In both case if we have single pod running then we are able to login into keycloak console but when we make it to 2 pods, then we are not able to login and in logs we get below error, continuously:
type=LOGIN_ERROR, realmId=<> clientId=null, userId=null, ipAddress=<>, error=expired_code, restart_after_timeout=true, authSessionParentId=eb97b946-7758-4f92-a4c1-f0113bd63539, authSessionTabId=KlUgAIpdnpc
  Can someone help to understand what is causing this? Also, need help to understand whether we should use statefulset or deployment?

Thanks.

**Comment by sschu** [comment]
This sounds like your pods cannot form a cluster successfully. Thus is not related to using a stateful set or a deployment. We are using stateful sets to run Keycloak (and the operator too).
